src/factors/Inertial/IMUDeltaFactor.jl

```diff
@@ -0,0 +1,467 @@
+    θ = norm(θ⃗)
+    if θ ≈ 0
+        return 1/2*SMatrix{3,3,T}(I)
+    else
+        u = θ⃗/θ
+        sθ, cθ = sincos(θ)
+        uₓ = skew(u)
+        return 1/2*SMatrix{3,3,T}(I) + (θ - sθ)/θ^2 * uₓ + (cθ + 1/2*θ^2 - 1)/θ^2 * uₓ^2
+    end
+end
+
+function Manifolds.exp(M::IMUDeltaGroup, X::ArrayPartition{T}) where T<:Real
+    θ⃗ₓ = X.x[1] # ωΔt
+    ρ = X.x[2]  # vΔt
+    ν = X.x[3]  # aΔt
+    Δt = X.x[4]
+
+    # ωΔt = vee(θ⃗ₓ)
+    θ⃗ = SA[θ⃗ₓ[3,2]; θ⃗ₓ[1,3]; θ⃗ₓ[2,1]]
+
+    P = _P(θ⃗)
+    Q = _Q(θ⃗)
+
+    q = ArrayPartition(
+        exp(θ⃗ₓ),
+        Q*ν,
+        Q*ρ + P*ν*Δt,
+        Δt,
+    )
+    return q
+end
+
+function Manifolds.exp(M::IMUDeltaGroup, p::ArrayPartition{T}, X::ArrayPartition{T}) where T<:Real
+    q = exp(M, X)
+    return Manifolds.compose(M, p, q)
+end
+
+function Manifolds.log(M::IMUDeltaGroup, p)
+    ΔR = p.x[1]
+    Δv = p.x[2]
+    Δp = p.x[3]
+    Δt = p.x[4]
+
+    Rv = RotationVec(ΔR)
+    θ⃗ = SA[Rv.sx, Rv.sy, Rv.sz]
+
+    P = _P(θ⃗)
+    Q = _Q(θ⃗)
+    iQ = inv(Q)
+    
+    return ArrayPartition(
+        log_lie(SpecialOrthogonal(3), ΔR), # θ⃗ₓ
+        iQ*(Δp - P*iQ*Δv*Δt), # ρ 
+        iQ*Δv, # ν
+        Δt
+    )
+end
+
+# TODO consolidate with Manifolds notation 
+# function Manifolds.log(M::IMUDeltaGroup, p::SMatrix{5,5,T}, q::SMatrix{5,5,T})
+#     qinvp = Manifolds.compose(M, inv(M, q), p)
+#     .....
+# end
+
+# compute the expected delta from p to q on the IMUDeltaGroup
+# ⊟ = boxminus
+function boxminus(::IMUDeltaGroup, p, q; g⃗ = SA[0,0,9.81]) 
+    Rᵢ = p.x[1]
+    vᵢ = p.x[2]
+    pᵢ = p.x[3]
+    tᵢ = p.x[4]
+
+    Rⱼ = q.x[1]
+    vⱼ = q.x[2]
+    pⱼ = q.x[3]
+    tⱼ = q.x[4]
+
+    Δt = tⱼ - tᵢ
+    ΔR = Rᵢ'Rⱼ
+    #TODO another possible mistake in paper as it does not include pᵢ
+    Δp = Rᵢ' * (pⱼ - pᵢ - vᵢ * Δt - 1/2 * g⃗ * Δt^2)
+    Δv = Rᵢ' * (vⱼ - vᵢ - g⃗ * Δt)
+
+    return ArrayPartition(
+        ΔR,
+        Δv,
+        Δp,
+        Δt
+    )
+end
+
+function ominus(M::IMUDeltaGroup, p, q)
+    log(M, Manifolds.compose(M, inv(M, q), p))
+end
+
+# small adjoint ad
+function adjointMatrix(::IMUDeltaGroup, X::ArrayPartition{T}) where T
+    θ⃗ₓ = X.x[1]
+    ρ = X.x[2]
+    ν = X.x[3]
+    IΔt = SMatrix{3,3,T}(X.x[4]*I)
+
+    ρₓ = skew(ρ)
+    νₓ = skew(ν)
+    
+    m0 = zeros(3,3) 
+    v0 = zeros(3)
+
+    return SMatrix{10,10,T}([
+         θ⃗ₓ  -IΔt  ρₓ   ν;
+         m0    θ⃗ₓ  νₓ  v0;
+         m0    m0  θ⃗ₓ  v0;
+              zeros(1,10);
+    ])
+
+end
+
+# Adjoint Ad
+function AdjointMatrix(::IMUDeltaGroup, p::ArrayPartition{T}) where T
+    ΔR = p.x[1]
+    Δv = p.x[2]
+    Δp = p.x[3]
+    Δt = p.x[4]
+
+    Δvₓ = skew(Δv)
+    pmvtₓ = skew(Δp - Δv*Δt)
+
+    m0 = zeros(3,3) 
+    v0 = zeros(3)
+
+    return SMatrix{10,10,T}([
+        ΔR  -ΔR*Δt  pmvtₓ*ΔR  Δv;
+        m0      ΔR    Δvₓ*ΔR  v0;
+        m0      m0        ΔR  v0;
+        zeros(1,9)             1;
+    ])
+
+end
+
+# right Jacobian
+function Jr(M::IMUDeltaGroup, X; order=3)
```

Comment:
This looks like a matrix form of `differential_exp_argument` from ManifoldDiff.jl. More specifically, `differential_exp_argument_lie_approx!`. Having it in matrix form makes sense though for covariance propagation, which can be done using matrix multiplication instead of repeated calls to `differential_exp_argument_lie_approx!`.